source/Engine.cpp

```diff
@@ -1248,10 +1248,22 @@ void Engine::MoveShip(const shared_ptr<Ship> &ship)
 	// Bail out if the ship just died.
 	if(ship->ShouldBeRemoved())
 	{
+		// Regular ships are silent about being removed.
+		if(!ship->IsSpecial())
+			return;
+		
 		// Make sure this ship's destruction was recorded, even if it died from
 		// self-destruct.
 		if(ship->IsDestroyed())
 			eventQueue.emplace_back(nullptr, ship, ShipEvent::DESTROY);
+		// The ship was able to flee. Carried ships only flee with the parent.
+		else if(!ship->CanBeCarried() || !ship->GetParent() || ship->GetParent()->ShouldBeRemoved())
+		{
+			eventQueue.emplace_back(nullptr, ship, ShipEvent::FLEE);
+			for(auto &bay : ship->Bays())
```

Comment:
`const Ship::Bay` isn't too long to write :)

Reply:
My style preference,  I used auto because it's an inner class of another class.
Not idea about the official stance regarding this.

Reply:
http://endless-sky.github.io/styleguide/styleguide.xml#C++11 has a bit about use of `auto`